.github/workflows/e2e-test.yaml

```diff
@@ -0,0 +1,76 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+name: End to End test
+permissions:
+  contents: read
+on:
+    push:
+    # Run on branch/tag creation
+    create:
+    # Run on pull requests
+    pull_request:
+  
+      
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+env:
+  CARGO_TERM_COLOR: always
+jobs:
+  e2e-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create Astarte Cluster
+        id: astarte
+        uses: astarte-platform/astarte-cluster-action@v1
+        with:
+          astarte_version: "1.1.1"
+      - name: Checkout sources
+        uses: actions/checkout@v3
+      - name: Install interface
+        run: |
+          astartectl realm-management interfaces sync $GITHUB_WORKSPACE/tests/e2etest/GenericTest/src/main/resources/standard-interfaces/*.json --non-interactive
+          astartectl realm-management interfaces ls
+      - name: Register device
+        run: |
+          echo "E2E_REALM=test" >> $GITHUB_ENV
+          DEVICE_ID=$(astartectl utils device-id generate-random)
+          echo "E2E_DEVICE_ID=$DEVICE_ID" >> $GITHUB_ENV
+          CREDENTIALS_SECRET=$(astartectl pairing agent register --compact-output -- "$DEVICE_ID")
+          echo "E2E_CREDENTIALS_SECRET=$CREDENTIALS_SECRET" >> $GITHUB_ENV
+          echo "Java_LOG=debug" >> $GITHUB_ENV
+          TOKEN=$(astartectl utils gen-jwt appengine)
+          echo "E2E_TOKEN=$TOKEN" >> $GITHUB_ENV
+          echo "E2E_API_URL=https://api.autotest.astarte-platform.org" >> $GITHUB_ENV
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Generate SSL certificates
+        run: |       
+          sudo openssl s_client -connect api.autotest.astarte-platform.org:443 -showcerts </dev/null 2>/dev/null|openssl x509 -outform PEM > ${GITHUB_WORKSPACE}/astarte-certificates.pem
+          openssl x509 -outform der -in ${GITHUB_WORKSPACE}/astarte-certificates.pem -out ${GITHUB_WORKSPACE}/astarte-certificates.der
+          echo $JAVA_HOME
+          KEYSTORE_PATH=$JAVA_HOME/lib/security/cacerts
+          sudo keytool -import -noprompt -trustcacerts -alias "astarteroot-cert" -file ${GITHUB_WORKSPACE}/astarte-certificates.der -keystore $KEYSTORE_PATH -storepass changeit
+          sudo keytool -keystore $KEYSTORE_PATH -storepass changeit -list | grep astarteroot-cert
```

Comment:
This could probably be simplified, I don't think you need to generate a new certificate. A valid certificate is already generated at the `/usr/local/share/ca-certificates/astarte-autotest.crt` path by the `astarte-cluster-action`. You should be able to import this one directly in the jvm key store.

Reply:
I had a lot of problems when I tried to create an end-to-end test for Java. The problem was related to SSL certification because our master branch does not have a flag to avoid SSL errors. I found a solution to fix the error. The solution was to add the SSL key to the JVM.